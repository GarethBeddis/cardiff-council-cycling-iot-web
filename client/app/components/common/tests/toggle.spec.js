import React from 'react'
import { shallow } from 'enzyme'

import Toggle from '../toggle'

describe('Toggle Component', () => {
  it('Checked Implementation renders without crashing', () => {
    shallow(<Toggle checked />)
  })
  it('Default Checked implementation renders without crashing', () => {
    shallow(<Toggle defaultChecked />)
  })
  // Snapshot Tests
  describe('Snapshots', () => {
    it('Checked toggle matches snapshot', () => {
      let wrapper = shallow(<Toggle checked />)
      expect(wrapper).toMatchSnapshot()
    })
    it('Default checked toggle matches snapshot', () => {
      let wrapper = shallow(<Toggle defaultChecked />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
