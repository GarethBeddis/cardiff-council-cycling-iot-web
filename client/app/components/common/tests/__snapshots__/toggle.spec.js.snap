// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Component Snapshots Checked toggle matches snapshot 1`] = `
<label
  className="toggle"
>
  <input
    checked={true}
    type="checkbox"
  />
  <div
    className="slider"
  >
    <div
      className="gradient"
    />
    <div
      className="switch"
    />
  </div>
</label>
`;

exports[`Toggle Component Snapshots Default checked toggle matches snapshot 1`] = `
<label
  className="toggle"
>
  <input
    defaultChecked={true}
    type="checkbox"
  />
  <div
    className="slider"
  >
    <div
      className="gradient"
    />
    <div
      className="switch"
    />
  </div>
</label>
`;
